pkg/controllers/job/job_controller.go

```diff
@@ -513,16 +525,32 @@ func (cc *jobcontroller) handleJobError(queue workqueue.TypedRateLimitingInterfa
 		req.Namespace, req.JobName, err)
 }
 
-func (cc *jobcontroller) cleanupDelayActions(jobKey string) {
+// cleanupDelayActions cleans up delayed actions
+// After a delayed action is executed, other delayed actions of the same type need to be cleaned up to avoid duplicate execution
+// Parameters:
+//   - currentDelayAction: the delayed action that has just been executed
+//
+// Implementation logic:
+//  1. Get the type of current delayed action (Job level, Task level or Pod level)
+//  2. Iterate through all delayed actions under this Job
+//  3. If the delayed action type matches, cancel it and remove from the map
+//
+// Usage scenarios:
+//   - When a Pod failure triggers Job termination, need to cancel other Job delayed actions under this Job
+func (cc *jobcontroller) cleanupDelayActions(currentDelayAction *delayAction) {
 	cc.delayActionMapLock.Lock()
 	defer cc.delayActionMapLock.Unlock()
 
-	if m, exists := cc.delayActionMap[jobKey]; exists {
+	actionType := GetActionType(currentDelayAction.action)
+
+	if m, exists := cc.delayActionMap[currentDelayAction.jobKey]; exists {
 		for _, delayAct := range m {
```

Comment:
Don't we need to check if the PodName of currentDelayAction equals the PodName of delayAction?